### DISCUSSION TITLE
Recommendations for Image Preloading

### DISCUSSION THREAD

**Discussion by andrskr** [ascorbic/unpic-img#707] [Q&A]
Hi everyone,

I'm currently implementing a responsive hero image using the **unpic-img** library and looking for recommendations on how to effectively preload the various image variants based on media queries. My goal is to optimize loading times while ensuring that the appropriate image is displayed for different screen sizes.

### Context:
- **Use Case**: I have a hero image that needs to adapt to various screen sizes and resolutions.
- **Approach**: I want to use the `<link rel="preload">` method as described in the [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/rel/preload).

### Questions:
- Are there any specific considerations or best practices when using this library for responsive image loading?
- Any additional tips or examples that you can share to optimize image loading using **unpic-img**?
- According to the doc, I can use something called `transformUrl` but I can find a utility that will generate a **list** of URLs similar to what the `Image` component do, when the end `img` has a `srcSet` property defined with all the sizes and URLs.
<img width="720" alt="Screenshot 2024-10-22 at 3 40 41 PM" src="https://github.com/user-attachments/assets/a87fb57b-fcc9-4433-a9ea-c6c06139905e">
<img width="719" alt="Screenshot 2024-10-22 at 3 44 43 PM" src="https://github.com/user-attachments/assets/83e007db-adbd-4009-ad37-0090341fb246">

I appreciate any insights or suggestions you might have!

Thank you!

**Comment by plokhovd** [comment]
I stumbled upon this post dealing with the same question.

Ended up using
- transformProps from '@unpic/core' to get srcset
- preload from react
